Source/Scene/Primitive.js

```diff
@@ -798,6 +811,7 @@ define([
                     }, function(error) {
                         that._error = error;
                         that._state = PrimitiveState.FAILED;
+                        this._readyPromise.resolve(this);
```

Comment:
Throughout `Primitive.js`, can we resolve this in `frameState.afterRender` so the called function can not interfere with the render loop?  This is done in `Model.js`.  If we do this a lot, perhaps we'll introduce something like `frameState.resolveAfterRender(promise)`.